atd-vzd/data_model_readme.md

```diff
@@ -0,0 +1,369 @@
+```
+
+#### 4. CRIS user updates the crash lat/lon and road type
+
+```sql
+update cris_crashes set latitude = 1, longitude = 1, road_type_id = 5 where crash_id = 9999999;
+```
+
+Observe that the lat/lon changes are not reflected by querying the `crashes` table because they are overridden by VZ values.
+
+#### 5. CRIS user updates a unit type 
+
+```sql
+update cris_units set unit_type_id = 4 where unit_id = 987654322;
+```
+
+#### 6. VZ user adds a custom lookup value and uses it
+
+```sql
+insert into unit_types (id, description) values (90001, 'custom unit type');
+update vz_units set unit_type_id = 90001 where unit_id = 987654322;
+```
+
+#### 7. Create a query that demonstrates the correct source of truth when crashes and units have edits from both CRIS and the VZ user
+
+The truth is stored in the `crashes` and `units` table. So we can just query them as needed. These queries also construct the `unique_unit_types` column, per the requirements. In reality we would establish dedicated views for computed columns that can be efficiently calculated on the fly.
+
+```sql
+-- 7a. Query for a single crash by ID
+select
+    crashes.*,
+    road_types.description as road_type_description,
+    unit_types.unique_unit_types
+from
+    crashes
+    left join road_types on crashes.road_type_id = road_types.id
+    left join (
+        select
+            crash_id,
+            array_agg(distinct unit_types.description) as unique_unit_types
+        from
+            units
+            left join unit_types on unit_types.id = units.unit_type_id
+        group by
+            crash_id
+    ) unit_types on unit_types.crash_id = crashes.crash_id
+    where crashes.crash_id = 9999999;
+
+-- 7b. query for 100k crashes
+select
+    crashes.*,
+    road_types.description as road_type_description,
+    unit_types.unique_unit_types
+from
+    crashes
+    left join road_types on crashes.road_type_id = road_types.id
+    left join (
+        select
+            crash_id,
+            array_agg(distinct unit_types.description) as unique_unit_types
+        from
+            units
+            left join unit_types on unit_types.id = units.unit_type_id
+        group by
+            crash_id
+    ) unit_types on unit_types.crash_id = crashes.crash_id
+    limit 100000;
+```
+
+#### 8. Create a query/view that powers a simplified version of the locations table  by calculating total number of units per location 
+
+```sql
+-- this is the top 100 locations ranked by # of crashes 🚀
+select
+    atd_txdot_locations.location_id,
+    count(crashes.crash_id) count_of_crashes
+from
+    atd_txdot_locations
+    left join crashes ON crashes.location_id = atd_txdot_locations.location_id
+group by
+    (atd_txdot_locations.location_id)
+order by
+    count_of_crashes DESC
+limit 100;
+```
+
+*Side note: it would be a nice performance optimization to use integers instead of strings for location IDs.*
+
+#### 9. Add a new editable column to crashes
```

Comment:
Either a small bug or I didn't run the SQL correctly, but I didn't see the default `TRUE` values coming through for the new columns in the `crashes` table.

Reply:
+1, i think john would have needed to add `default true` to the `alter` statement for the `crashes` table as well, i think whats happening is the values there aren't gonna get updated until an actual update from cris happens and forces the default true val into the `crashes` record

Reply:
ah yes, defaulting to `TRUE` on the `cris` table really didn't make sense without also doing a backfill. thanks for flagging that.